src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -684,8 +684,14 @@ public function dumpFile($filename, $content)
         // when the filesystem supports chmod.
         $tmpFile = $this->tempnam($dir, basename($filename));
 
-        if (false === @file_put_contents($tmpFile, $content)) {
-            throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
+        if (\is_resource($content)) {
+            $expectedSize = fstat($content)['size'] - ftell($content);
```

Comment:
Is this really portable? I think some streams cannot be stat'ed.

Reply:
Any particular you have in mind so we can add test cases? The stream interface requires implementing stat, don't know if there are some that don’t do it.

Reply:
I think e.g. the http stream wrapper when the remote server uses chunked encoding and doesn't return a content-length. Sure, stats would return a size, but a wrong one for the use case here.
The only way would be to buffer the full response in memory then write it, but that's break streaming of resources...